msal/src/test/java/com/microsoft/identity/nativeauth/NativeAuthPublicClientApplicationKotlinTest.kt

```diff
@@ -2291,5 +2394,33 @@ class NativeAuthPublicClientApplicationKotlinTest : PublicClientApplicationAbstr
         val result = application.signUp(invalidUsername)
         assertTrue(result is SignUpError)
         assertTrue((result as SignUpError).isInvalidUsername())
+
+        checkSafeLogging()
+    }
+
+    private fun checkSafeLogging() {
+        val piiTrueToCheck = listOf(
+            "password:","password=",
+            "code:","code=",
+            "continuationToken:","continuationToken="
+        )
+        val piiFalseToCheck = listOf(
+            "username:", "username="
+        )
+
+        val elementsToCheck = piiTrueToCheck.toMutableList()
+
+        if (!allowPII) {
+            elementsToCheck.addAll(piiFalseToCheck)
+        }
+
+        elementsToCheck.forEach { element ->
+            verifyLogDoesNotContain("Command", element)
+        }
+
+        Logger.getInstance().setExternalLogger(null)
```

Comment:
Isn't this the same as what's been done in `cleanup()`?

Reply:
Yes. This is the legacy code. However, setExternalLogger(null) cannot be used as an alternative to new method of removeExternalLogger

Reply:
Why not?

Reply:
```
    public synchronized void setExternalLogger(final ILoggerCallback externalLogger) {
        if (externalLogger == null) {
            return;
        }
```
Because it doesn't look like that's the recommended method.